tiledb/type/range/range.h

```diff
@@ -61,14 +62,28 @@ class Range {
    * The range is stored as a sequence of bytes with manual memory layout. The
    * memory layout is different for fixed-size and variable-size types.
    *
+   * The type is tdb::pmr::pmr_vector<uint8_t> instead of
```

Comment:
@davisp Are there any reasons we disabled the copy constructors in tdb::pmr::vector? Are we worried the copy constructor that doesn't take a memory tracker will not do the right thing and we might have some allocator mixup?

Reply:
There are two copy constructors. One takes an allocator and one doesn't. The one that takes an allocator is available for use.

The copy constructor that doesn't take an allocator is disabled. That choice was so that as we changed from std::vector to std::pmr::vector, we'd not miss any place where we used copy constructors because the compiler would catch the issue and force us to update those call sites with the allocator argument.

Reply:
* I restored the move constructor and move-assign operator in `tdb::pmr::vector`.
* I manually implemented the copy constructor and copy-assign operator in `Range` to avoid breaking other uses.